pkg/reconcile/reconcilers_test.go

```diff
@@ -307,9 +316,90 @@ func TestPodSecurityContext(t *testing.T) {
-	namespace := "service-test"
-	reaperName := "test-reaper"
+func TestReaperDeploymentStrategy(t *testing.T) {
+	// Test to ensure that if we set DeploymentStrategy to Rolling then it gets set back to Recreate by reconcile()
+	// Reaper test resource
+	testReaper := newReaperWithCassandraBackend("default", "test-reaper")
+	runAsUser := int64(8675309)
+	testReaper.Spec.Image = "test/reaper:latest"
+	testReaper.Spec.PodSecurityContext = &corev1.PodSecurityContext{
+		RunAsUser: &runAsUser,
+	}
+	// Create CassDC
+	testCassDC := newCassDC("default", "dc1")
+
+	// Test Scheme
+	localSchemeBuilder := runtime.SchemeBuilder{
+		api.AddToScheme,
+		appsv1.AddToScheme,
+		cassdcapi.AddToScheme,
+	}
+	AddToScheme := localSchemeBuilder.AddToScheme
+	testScheme := scheme.Scheme
+	AddToScheme(testScheme)
+
+	// Mock client
+	k8sClient := fake.NewFakeClientWithScheme(testScheme, testCassDC.DeepCopy())
+
+	// Other fixtures
+	testSecretsmngr := NewSecretsManager()
+	testctx := context.Background()
+	testStatusManager := &status.StatusManager{Client: k8sClient}
+	testLogger := mocks.NewTabLogger()
+
+	// Test reconcile deployment
+	r := defaultReconciler{
+		k8sClient,
+		testScheme,
+		testSecretsmngr,
+	}
+	req := ReaperRequest{
+		testReaper,
+		testLogger,
+		testStatusManager,
+	}
+	r.ReconcileDeployment(testctx, req)
+
+	// Ensure Reaper is deploying with RecreateStrategy
+	deploymentList := &appsv1.DeploymentList{}
+	if err := k8sClient.List(testctx, deploymentList, client.MatchingLabels{"app.kubernetes.io/managed-by": "reaper-operator"}); err != nil {
+		t.Log(err)
+		assert.FailNow(t, "Failed to get Reaper instance")
+	}
+	reaperDeployment := deploymentList.Items[0].DeepCopy()
+	assert.True(t, len(deploymentList.Items) == 1)                                               // Only one deployment should be observed
+	assert.True(t, reaperDeployment.Spec.Strategy.Type == appsv1.RecreateDeploymentStrategyType) // Deployment type is right.
+
+	// Change deploymentStrategy to RollingUpdate
+	reaperDeploymentPatch := client.MergeFrom(reaperDeployment.DeepCopy())
+	reaperDeployment.Spec.Strategy = appsv1.DeploymentStrategy{
+		Type: appsv1.RollingUpdateDeploymentStrategyType,
+	}
+	if err := r.Patch(testctx, reaperDeployment, reaperDeploymentPatch); err != nil {
+		t.Log(err)
+		assert.FailNow(t, "Failed to patch deployment with appsv1.RecreateDeploymentStrategyType")
+	}
+
+	// Check that the Deployment strategy was changed to Rolling
+	// TODO - this doesn't currently pass. The strategy doesn't appear to get changed by our patching process.
+	if err := k8sClient.List(testctx, deploymentList, client.MatchingLabels{"app.kubernetes.io/managed-by": "reaper-operator"}); err != nil {
+		t.Log(err)
+		assert.FailNow(t, "Failed to get Reaper Deployment")
+	}
+	assert.True(t, deploymentList.Items[0].Spec.Strategy.Type == appsv1.RollingUpdateDeploymentStrategyType)
+
+	// Reconcile - expect to see DeploymenStrategyType change back to Recreate
```

Comment:
-> envtest ones also. These are just a couple of lines to reaper_controller_test.go